pkgs/by-name/ya/yazi-unwrapped/update.sh

```diff
@@ -35,7 +35,11 @@ pushd "$SCRIPT_DIR"
 # Build date
 sed -i 's#env.VERGEN_BUILD_DATE = "[^"]*"#env.VERGEN_BUILD_DATE = "'"${latestBuildDate}"'"#' package.nix
 
+# Cargo.lock
+curl -L -o 'Cargo.lock' "https://raw.githubusercontent.com/sxyazi/yazi/$tagName/Cargo.lock"
+git add 'Cargo.lock'
+
 # Hashes
-cargoHash=$(nix-prefetch "{ sha256 }: (import $NIXPKGS_DIR {}).yazi-unwrapped.cargoDeps.overrideAttrs (_: { outputHash = sha256; })")
-sed -i -E 's#\bcargoHash = ".*?"#cargoHash = "'"$cargoHash"'"#' package.nix
+# cargoHash=$(nix-prefetch "{ sha256 }: (import $NIXPKGS_DIR {}).yazi-unwrapped.cargoDeps.overrideAttrs (_: { outputHash = sha256; })")
+# sed -i -E 's#\bcargoHash = ".*?"#cargoHash = "'"$cargoHash"'"#' package.nix
```

Comment:
Perhaps we can parse the `Cargo.lock` file and see if there are any "git+" dependencies to know whether to use `cargoLock` or `cargoHash`. What do you think?

Reply:
Great idea. Find it seems easy, but switch between `cargoHash` and `cargoLock` may be difficult. I'll try to implement it.

Reply:
I have implemented it by ast-grep.